x/stakeibc/types/message_register_community_pool_rebate.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v19/utils"
+)
+
+const TypeMsgRegisterCommunityPoolRebate = "register_community_pool_rebate"
+
+var _ sdk.Msg = &MsgRegisterCommunityPoolRebate{}
+
+func NewMsgRegisterCommunityPoolRebate(
+	creator string,
+	chainId string,
+	rebatePercentage sdk.Dec,
+	liquidStakeAmount sdkmath.Int,
+) *MsgRegisterCommunityPoolRebate {
+	return &MsgRegisterCommunityPoolRebate{
+		Creator:           creator,
+		ChainId:           chainId,
+		RebatePercentage:  rebatePercentage,
+		LiquidStakeAmount: liquidStakeAmount,
+	}
+}
+
+func (msg *MsgRegisterCommunityPoolRebate) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRegisterCommunityPoolRebate) Type() string {
+	return TypeMsgRegisterCommunityPoolRebate
+}
+
+func (msg *MsgRegisterCommunityPoolRebate) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgRegisterCommunityPoolRebate) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRegisterCommunityPoolRebate) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+	if err := utils.ValidateAdminAddress(msg.Creator); err != nil {
+		return err
+	}
+	if msg.ChainId == "" {
+		return errors.New("chain ID must be specified")
+	}
+	if msg.RebatePercentage.IsNil() || msg.RebatePercentage.LT(sdk.ZeroDec()) || msg.RebatePercentage.GTE(sdk.OneDec()) {
+		return errors.New("invalid rebate percentage, must be between [0, 1)")
+	}
+	if msg.LiquidStakeAmount.IsNil() || msg.LiquidStakeAmount.LTE(sdkmath.ZeroInt()) {
```

Comment:
I think we want to allow setting amount to 0 if CP redeems everything - if so change this check to:

`if msg.LiquidStakeAmount.IsNil() || msg.LiquidStakeAmount.LT(sdkmath.ZeroInt())`

Reply:
good call - I'll update so that you can set it to 0, but that it will also delete the rebate in that case

Reply:
wait actually, the way it's implemented, 0 is allowed

Reply:
lgtm